### ISSUE TITLE
Add Optional thisContext and fix error for new typescript 2.1.5

### ISSUE THREAD

**Issue by ricklove** [notenoughneon/typed-promisify#2]
Thanks for the implementation!

Here is a quick improvement I needed that allows preserving the this context when needed:

```
export type Nodeback<T> = (err: any, result: T) => void;

export function promisify<T>(f: (cb: Nodeback<T>) => void, thisContext: any): () => Promise<T>;
export function promisify<A, T>(f: (arg: A, cb: Nodeback<T>) => void, thisContext: any): (arg: A) => Promise<T>;
export function promisify<A, A2, T>(f: (arg: A, arg2: A2, cb: Nodeback<T>) => void, thisContext: any): (arg: A, arg2: A2) => Promise<T>;
export function promisify<A, A2, A3, T>(f: (arg: A, arg2: A2, arg3: A3, cb: Nodeback<T>) => void, thisContext: any): (arg: A, arg2: A2, arg3: A3) => Promise<T>;
export function promisify<A, A2, A3, A4, T>(f: (arg: A, arg2: A2, arg3: A3, arg4: A4, cb: Nodeback<T>) => void, thisContext: any): (arg: A, arg2: A2, arg3: A3, arg4: A4) => Promise<T>;
export function promisify(f: any, thisContext: any = null) {
    return function () {
        let args = Array.prototype.slice.call(arguments);
        return new Promise((resolve, reject) => {
            args.push((err: any, result: any) => err !== null ? reject(err) : resolve(result));
            f.apply(thisContext, args);
        });
    }
}
```

Oh, I also fixed the error by moving the args up:

```
    return function () {
        // To Here:
        let args = Array.prototype.slice.call(arguments);
        return new Promise((resolve, reject) => {
            // From Here
            args.push((err: any, result: any) => err !== null ? reject(err) : resolve(result));
            f.apply(thisContext, args);
        });
    }
```

**Comment by ricklove** [comment]
#3 
I made this into a pull request and confirmed the tests pass.

**Comment by notenoughneon** [comment]
Fixed by #3
